New UI v3.1

WanderGomes opened the pull request:
Improvements to the CSS, HTML, and JS styles to reflect the new design of version 3.1.

Here’s a brief summary of the changes in the pull request:

- Updated all CSS files in the “new-ui” folder to match the new design.
- Added a new icon font “filesender-icons” based on Marleen’s icons. We used the “iconmoon.io” tool to generate the icon font files and versioned the icon selection in the “iconmoon-selection.json” file. Future updates can easily import this file to add more icons.
- Updated some template files to reflect the new design, especially where the screen structure changed.
Added new terms to the language files.

monkeyiq commented:
I will skip the language files. Changes made to them will be overridden with the data from poeditor when bulk imports happen.

monkeyiq commented:
I notice that the encryption option is now hidden under "settings". I know that some people at some NRENs have extremely strong opinions about having encryption hidden. 

![filesender31-upload](https://github.com/user-attachments/assets/23d59249-e56a-4ad5-832f-9b4bc598430f)

monkeyiq commented:
We should include the copyright notice and license for the bundled font as per https://fonts.google.com/specimen/Nunito+Sans/license

  "Original or Modified Versions of the Font Software may be bundled, redistributed and/or sold with any software, provided that each copy contains the above copyright notice and this license. "

monkeyiq commented:
It looks like some files in www/fonts/FileSender-Icons are using a tool IcoMoon. "Generated by IcoMoon". If these are using some icon packs or whatnot from IcoMoon it would be nice to know. On the other hand if these are all original works for the project then know that is also useful.

madsi1m commented:
> I notice that the encryption option is now hidden under "settings". I know that some people at some NRENs have extremely strong opinions about having encryption hidden.
> 
> ![filesender31-upload](https://private-user-images.githubusercontent.com/387108/379503836-23d59249-e56a-4ad5-832f-9b4bc598430f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3MzU2MDgsIm5iZiI6MTcyOTczNTMwOCwicGF0aCI6Ii8zODcxMDgvMzc5NTAzODM2LTIzZDU5MjQ5LWU1NmEtNGFkNS04MzJmLTliNGJjNTk4NDMwZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNFQwMjAxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYWMzMDkzMzY3YmY4MzA2MWMwZjVkNDkzNmUxMzJmZDk4ZTNkNWZjNTkxODA1MWY4YjlhMzEyM2ZjYjVkNTg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Seef2MzBHbhZozx_AyaHlTkDfL59Mh3-ETvlbS4rFPk)

We are one of those NRENs that have extremely strong opinions about having encryption hidden.
This is a feature that makes FileSender stand out so it should live front and center with neon lights and massive arrows pointing at it.

Please unhide it.

efef commented:
@madsi1m when encryption is important, maybe it would be better to enforce it ?

monkeyiq commented:
The issue here is if encryption is hidden under settings then many users might not even notice that it is possible.

monkeyiq commented:
For 2.x series there are the following options to allow enforced encryption.
```
$config['encryption_mandatory'] = true;
$config['encryption_mandatory_with_generated_password'] = true;
```

monkeyiq commented:
It seems there is a regression with `encryption_mandatory_with_generated_password` in both development3 and this PR.

monkeyiq commented:
I have created an issue to track encryption_mandatory_with_generated_password https://github.com/filesender/filesender/issues/2057

madsi1m commented:
not to derail the convo, @WanderGomes please unhide the encryption

monkeyiq commented:
IIRC in one of the previous ui rewrites SURF was also very keen for the encryption option to not be hidden. 

This is not so much of a deal breaker as I can also optionally unhide it later with the addition of yet another configuration option to "lift" it.

monkeyiq commented:
Thank you and the UI team for this update to the UI!

I guess a summary of things so far

* Add copyright notice and this license for introduced font
* Add license + provenance information for new graphics. This may be as simple as a README file in www/fonts/FileSender-Icons mentioning that the icons are original works created RNP and are licensed under the same "BSD-3-Clause license" as the rest of FileSender.
* Unless the FileSender board is happy to download stylesheets from `fonts.googleapis.com` we should include them on a deployment (ie, in the PR itself). The past policy has been _very much_ against adding loads from third party sites. 
* Updates to language files must be ported to poeditor. Items in this PR will be overridden by data from poeditor in the future.

Optional:

* Lift encryption option to main page. If making the encryption shown by default is seen as something to avoid by the UI design team then I can add a another new configuration option to allow sites who want to always show it continue to have it.
* I notice that the FileSender-Icons.svg can be loaded into FontForge. This provides a path forward to updating the generated font with FileSender images with FOSS in the future. Normally where possible a build script using FOSS tools for generating derived works would be expected in a PR to a FOSS project.

There are likely other configuration options that have been added in the last year to the UI3 which I will have to update post PR.

WebSpider commented:
Does this UI update 3.1 also mean that FileSender will postpone releasing 3.0 until this new 3.1 UI is absorbed?

WebSpider commented:
> IIRC in one of the previous ui rewrites SURF was also very keen for the encryption option to not be hidden.

I know at SURF they would prefer to see encryption not being hidden. If it didn't impose too many restrictions on the decrypting client, encryption would have been made default already.

monkeyiq commented:
@WebSpider good question about when this might be merged. I have no idea of this will be merged into an RC or what the plan is.

madsi1m commented:
> Does this UI update 3.1 also mean that FileSender will postpone releasing 3.0 until this new 3.1 UI is absorbed?

I hope not, that said we currently just follow the development3 branch and lock in certain commits, rather than waiting for official releases, this might change once v3 gets a stable full release.

> In general I lament that the UI work has been done in so much isolation.

100% feel the same way...

marl1v commented:
The initial wireframes of the new UI did not have the encryption hidden in advanced settings, however user feedback requested to simplify the upload flow. This is why, in the final UI, only basic information (file upload + way of transferring + recipient/message) is shown and all other information has been put under Settings in a fold out. The thought behind this is to show the user as little information/noise as possible when they start the transfer process. 
I understand the importance of the encryption and I think it would be no problem to put it back as a main component in the flow, like so:

<img width="906" alt="Screenshot 2024-11-08 at 08 59 57" src="https://github.com/user-attachments/assets/f2f53ef3-d817-4af5-b19f-9bcb9e60f69a">

monkeyiq commented:
Does this PR require changes to the CSP headers?

WanderGomes commented:
@monkeyiq Not at the moment, but if we use Google Fonts in the future, then yes.

In this version, I’m not using Google Fonts because the Nunito Sans font family has issues when embedded through it, as it doesn’t render correctly. That’s why I added the font file directly to the project. However, I’ve already prepared a feature in GUI.class.php to make it easy to add any Google Fonts in the future.

WanderGomes commented:
@monkeyiq @marl1v I’m going to move the encryption option out of the advanced settings.

WebSpider commented:
In v2 ui it is configurable in `transfer_options` if settings are available, what its default is, and if it should be in the Advanced section or the standard section.

Can this behaviour be mirrored to 3.1 ui?

WanderGomes commented:
@WebSpider  

> In v2 ui it is configurable in `transfer_options` if settings are available, what its default is, and if it should be in the Advanced section or the standard section.
> 
> Can this behaviour be mirrored to 3.1 ui?

In config.php, right?

We can do it like that, by setting where it’ll appear in the config.php.

WanderGomes commented:
> In v2 ui it is configurable in `transfer_options` if settings are available, what its default is, and if it should be in the Advanced section or the standard section.
> 
> Can this behaviour be mirrored to 3.1 ui?

In the config.php file, right? I didn’t modify this behavior, so we can do it like this, by setting where it will appear in the config.php.

WebSpider commented:
Yes, these options are in config.php

WanderGomes commented:
@monkeyiq @WebSpider I’ll make these changes and update this Pull Request throughout next week.

monkeyiq commented:
This issue with the font license is making sure we are doing what is needed 
```
Original or Modified Versions of the Font Software may be bundled, 
redistributed and/or sold with any software, provided that each copy 
contains the above copyright notice and this license. 
```

While they allow it to be in `appropriate machine-readable metadata fields` it is much simpler to know it is there if it is in an external text file in the font directory.

WanderGomes commented:
@monkeyiq  @WebSpider 

I just added two more commits to this pull request:

First commit: Adds the ability to toggle the encryption option between being displayed inside or outside the advanced settings. To configure this, simply add the following snippet to config.php:
php
Copy code
'encryption' => array(
    'available' => true,
    'advanced' => false,
    'default' => false
)

Second commit: Fixes step validation when switching between the "Get a Transfer Link" and "Send a Transfer via Email" options.

WanderGomes commented:
@monkeyiq @WebSpider 

I’ve updated the pull request with some improvements based on my email conversation with Olga:

- Changed the menu/header to use the compact FileSender logo.
- Removed the border-radius that was incorrectly applied to the logo in the menu.
- Made some style adjustments to the theme, header, and footer to use the correct logo versions for both light (default) and dark themes.

monkeyiq reviewed (changes requested):
I guess a summary of things so far

* Add copyright notice and this license for introduced font
* Add license + provenance information for new graphics. This may be as simple as a README file in www/fonts/FileSender-Icons mentioning that the icons are original works created RNP and are licensed under the same "BSD-3-Clause license" as the rest of FileSender.
* Unless the FileSender board is happy to download stylesheets from fonts.googleapis.com we should include them on a deployment (ie, in the PR itself). The past policy has been very much against adding loads from third party sites.
* Updates to language files must be ported to poeditor. Items in this PR will be overridden by data from poeditor in the future.
